semgrep-core/src/runner/Run_semgrep.ml

```diff
@@ -471,8 +476,10 @@ let semgrep_with_rules config ((rules, skipped_rules), rules_parse_time) =
            let file = target.In.path in
            let xlang = Xlang.of_string target.In.language in
            let rules =
-             Common.map
-               (fun r_id -> Hashtbl.find rule_table r_id)
+             (* Assumption: find_opt will return None iff a r_id
```

Comment:
IMO we should check the assumption and not just document it. In this case it's probably cheap to do so (we don't expect skippe_rules to be large)? If the assumption check fails, we can call `logger#error`.